veniq/baselines/semi/recommend.py

```diff
@@ -99,19 +102,27 @@ def _convert_ExtractionOpportunity_to_EMO(
     return (start_line_opportunity, start_line_opportunity + addit_lines_brackets)
 
 
-def check_format_validity(method_decl: List[str]) -> None:
+def check_input_method_format(method_decl: str) -> None:
     """
     Checks that the input method declaration is a syntactically
     correct Java method declaration.
     Raises exception otherwise.
     """
-    # TODO
-    pass
+    # check type
+    if type(method_decl) != str:
+        raise WrongInputToApi('Input should be a string with line breaks.')
+
+    # check it is a method declaration
+    class_interf_pattern = r"\s*(|(public|protected|private|static|)\s+)(class|interface)\s+[\w\W]+\s*"
```

Comment:
i do not like regexp here. It's too weak. Seems i can generate lots of example where it cannot work.

Reply:
@acheshkov what do u think?

Reply:
We can use java lang here to try to parse source code, something like this 

```
tokens = javalang.tokenizer.tokenize(you_source_code_string)
parser = javalang.parser.Parser(tokens)
maybe_parsed_method = parser.parse_member_declaration()
```